encoder.go

```diff
@@ -99,9 +99,17 @@ func (d *textDecoder) Decode(t string, value []byte) (driver.Value, error) {
 	v := string(value)
 	switch t {
 	case "Date":
-		return time.ParseInLocation(dateFormat, unquote(v), d.location)
+		uv := unquote(v)
+		if uv == "0000-00-00" {
+			return time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC), nil
```

Comment:
why UTC, I do not found in documentation, that it is in UTC.
there is only this in doc:
```
 The minimum value is output as 0000-00-00.
```
Maybe need to create date in d.location instead of UTC?